modules/command-engine/core/src/main/scala/surge/core/KafkaProducerActor.scala

```diff
@@ -92,64 +129,70 @@ class KafkaProducerActor(
       }(ExecutionContext.global)
   }
 
-  override def restart(): Unit = {}
-
-  override def start(): Unit = {
-    publisherActor ! ActorLifecycleManagerActor.Start
-
-    // todo: Register with reference to Controllable rather than ActorRef
-    // Register
-    val registrationResult =
-      signalBus.register(publisherActor, componentName = "kafka-producer-actor", restartSignalPatterns = restartSignalPatterns())
-
-    registrationResult.onComplete {
-      case Failure(exception) =>
-        log.error("KafkaProducerActor registration failed", exception)
-      case Success(done) =>
-        log.debug(s"KafkaProducerActor registration succeeded - ${done.success}")
+  override def restart(): Future[Ack] = {
+    for {
+      stopped <- stop()
+      started <- start(stopped)
+    } yield {
+      if (Option(stopped).isDefined && Option(started).isDefined) {
+        Ack()
+      } else {
+        throw new RuntimeException(s"Failed to restart $getClass")
+      }
     }
   }
 
-  override def stop(): Unit = {
-    publisherActor ! ActorLifecycleManagerActor.Stop
+  override def start(): Future[Ack] = {
+    implicit val askTimeout: Timeout = Timeout(TimeoutConfig.PublisherActor.askTimeout)
+
+    val result = publisherActor
+      .ask(ActorLifecycleManagerActor.Start)
```

Comment:
use mapTo[ActorLifecycleManagerActor.Ack]  instead of map and you won't have to worry about case _ and throwing an exception

Reply:
done